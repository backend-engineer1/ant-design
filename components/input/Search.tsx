import React from 'react';
import classNames from 'classnames';
import Input, { InputProps } from './Input';
import Icon from '../icon';

export interface SearchProps extends InputProps {
  onSearch?: (value: string) => any;
}

export default class Search extends React.Component<SearchProps, any> {
  static defaultProps = {
    prefixCls: 'ant-input-search',
  };
  input: any;
  onSearch = () => {
    const { onSearch } = this.props;
    if (onSearch) {
      onSearch(this.input.refs.input.value);
    }
    this.input.focus();
  }
  render() {
    const { className, prefixCls, ...others } = this.props;
    delete (others as any).onSearch;
    const searchSuffix = (
      <Icon
        className={`${prefixCls}-icon`}
        onClick={this.onSearch}
        type="search"
      />
    );
    return (
      <Input
        onPressEnter={this.onSearch}
        {...others}
        className={classNames(prefixCls, className)}
        suffix={searchSuffix}
        ref={node => this.input = node}
      />
    );
  }
}
