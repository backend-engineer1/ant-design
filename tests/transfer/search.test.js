import React, { Component } from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Search from '../../components/transfer/search';

describe('Search', () => {
  it('shows cross icon when user inputting', () => {
    const wrapper = mount(<Search value />);

    expect(mountToJson(wrapper)).toMatchSnapshot();

    wrapper.setProps({ value: 'a' });

    expect(mountToJson(wrapper)).toMatchSnapshot();
  });
});
