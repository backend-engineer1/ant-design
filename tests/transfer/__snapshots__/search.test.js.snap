exports[`Search shows cross icon when user inputting 1`] = `
<Search
  placeholder=""
  value={true}>
  <div>
    <Input
      autosize={false}
      disabled={false}
      onChange={[Function]}
      placeholder=""
      prefixCls="ant-input"
      type="text"
      value={true}>
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        type="text"
        value={true} />
    </Input>
    <span
      className="undefined-action">
      <Component
        type="search">
        <i
          className="anticon anticon-search" />
      </Component>
    </span>
  </div>
</Search>
`;

exports[`Search shows cross icon when user inputting 2`] = `
<Search
  placeholder=""
  value="a">
  <div>
    <Input
      autosize={false}
      disabled={false}
      onChange={[Function]}
      placeholder=""
      prefixCls="ant-input"
      type="text"
      value="a">
      <input
        className="ant-input"
        disabled={false}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder=""
        type="text"
        value="a" />
    </Input>
    <a
      className="undefined-action"
      href="#"
      onClick={[Function]}>
      <Component
        type="cross-circle">
        <i
          className="anticon anticon-cross-circle" />
      </Component>
    </a>
  </div>
</Search>
`;
